### DISCUSSION TITLE
Is there any way to swap the components ?

### DISCUSSION THREAD

**Discussion by echobinod** [GrapesJS/grapesjs#3836] [Q&A]
Hello @artf,
```
<div class="wrapper">
    <div class="component1">
    component 1
    </div>
    <div class="component2">
    component 2
    </div>
</div>
```
Here I want to swap component 1 and component 2, How can I achieve this ?

**Comment by artf** [comment]
You can [move](https://grapesjs.com/docs/api/component.html#move) components programmatically.

**Reply by echobinod** [reply]
```
const dest = editor.getWrapper();
editor.getSelected().move(dest, { at: 0 });
```

@artf how to move the component at specific #id or unique attribute with value ? The destination I need is at some element with #id

**Reply by iabhiyaan** [reply]
I think
```
editor.getSelected().move(dest, { at: editor.getSelected().index() + 1 });
Or, 
editor.getSelected().move(dest, { at: editor.getSelected().index() - 1 });
```

**Reply by artf** [reply]
you can also query your components with [`find`](https://grapesjs.com/docs/api/component.html#find), eg.
```js
const newDest = editor.getWrapper().find(`#some-id`)[0];
```

**Reply by echobinod** [reply]
Thanks @artf

**Reply by echobinod** [reply]
Thanks @iabhiyaan  that is what needed.
